lib/sidekiq/cron/job.rb

```diff
@@ -56,21 +56,23 @@ def enque! time = Time.now.utc
               nil
             end
 
-        if klass_const
-          if defined?(ActiveJob::Base) && klass_const < ActiveJob::Base
-            enqueue_active_job(klass_const)
-          else
-            enqueue_sidekiq_worker(klass_const)
-          end
-        else
-          if @active_job
-            Sidekiq::Client.push(active_job_message)
+        jid =
```

Comment:
you are setting `jid = ` and then in every branch of condition you are doing the same.

if this it the case then why not to make method 

```ruby
def enqueue_job_based_on_type
         klass_const =
             begin
               Sidekiq::Cron::Support.constantize(@klass.to_s)
             rescue NameError
               nil
             end
  if klass_const
            if defined?(ActiveJob::Base) && klass_const < ActiveJob::Base
              enqueue_active_job(klass_const).try :provider_job_id
            else
              enqueue_sidekiq_worker(klass_const)
            end
          else
            if @active_job
              Sidekiq::Client.push(active_job_message)
            else
              Sidekiq::Client.push(sidekiq_worker_message)
            end
          end
end
```
and then just do `jid = enqueue_job_based_on_type`